pkg/kubelet/metrics/metrics.go

```diff
@@ -47,6 +51,17 @@ const (
 	// Metrics keys of device plugin operations
 	DevicePluginRegistrationCountKey = "device_plugin_registration_count"
 	DevicePluginAllocationLatencyKey = "device_plugin_alloc_latency_microseconds"
+
+	// Metric keys for node config (these mirror the Assigned, Active, and LastKnownGood sources reported in Node.Status.Config)
+	AssignedConfigKey             = "node_config_assigned"
+	ActiveConfigKey               = "node_config_active"
+	LastKnownGoodConfigKey        = "node_config_last_known_good"
+	ConfigErrorKey                = "node_config_error"
+	ConfigNameLabelKey            = "node_config_name" // this is a fully resolved name, e.g. an API path or a URL, "Name" is short but a little confusing name for this key...
```

Comment:
@dashpole any naming suggestions for this?

Reply:
SelfLink is the URL representing an object in ObjectMeta.  Maybe "node_config_self_link"?  This should be equivalent to the self link for the configmap, right?

Reply:
@dashpole and I settled on node_config_source, as it accurately identifies what the corresponding value describes while being reasonably generic ("local", or a SelfLink, or maybe some URL in the future).